core/src/main/java/tc/oc/pgm/community/features/VanishManagerImpl.java

```diff
@@ -198,7 +205,13 @@ public void onUnvanish(PlayerVanishEvent event) {
 
   @EventHandler
```

Comment:
This event could be handled before the initial party is set to be a competitor, missing this logic. Make this happen on monitor